receiver/kubeletstatsreceiver/internal/kubelet/mem.go

```diff
@@ -21,4 +23,15 @@ func addMemoryMetrics(mb *metadata.MetricsBuilder, memoryMetrics metadata.Memory
 	recordIntDataPoint(mb, memoryMetrics.WorkingSet, s.WorkingSetBytes, currentTime)
 	recordIntDataPoint(mb, memoryMetrics.PageFaults, s.PageFaults, currentTime)
 	recordIntDataPoint(mb, memoryMetrics.MajorPageFaults, s.MajorPageFaults, currentTime)
+
+	if s.UsageBytes != nil {
+		if r.memoryLimit > 0 {
+			value := math.Min(float64(*s.UsageBytes)/float64(r.memoryLimit), 1)
```

Comment:
Why do we need to cap it at 1? Can't the usage go above the limit right before being killed?

Reply:
Yes. It comes down to whether we want to represent both limit and request as soft limits. Our earlier discussions had limit being a hard limit and request being a soft limit, but they could both be treated as soft limits if we want.